pool/src/pool.rs

```diff
@@ -111,816 +70,112 @@ where
-		self.search_stempool_spents(&output_ref.commit)
-			.or(self.pool.get_available_output(&output_ref.commit).map(|x| {
-				let tx_ref = x.source_hash().unwrap();
-				Parent::PoolTransaction { tx_ref }
-			}))
-			.or(self.stempool
-				.get_available_output(&output_ref.commit)
-				.map(|x| {
-					let tx_ref = x.source_hash().unwrap();
-					Parent::StemPoolTransaction { tx_ref }
-				}))
-			.or(self.search_blockchain_unspents(output_ref))
-			.or(self.search_pool_spents(&output_ref.commit))
-			.unwrap_or(Parent::Unknown)
+	/// TODO - not yet fully implemented
+	pub fn prepare_mineable_transactions(&self, num_to_fetch: u32) -> Vec<Box<Transaction>> {
+		self.transactions.clone()
 	}
 
-	// search_blockchain_unspents searches the current view of the blockchain
-	// unspent set, represented by blockchain unspents - pool spents, for an
-	// output designated by output_commitment.
-	fn search_blockchain_unspents(&self, output_ref: &OutputIdentifier) -> Option<Parent> {
-		self.blockchain.is_unspent(output_ref).ok().map(|_| {
-			match self.pool.get_blockchain_spent(&output_ref.commit) {
-				Some(x) => {
-					let other_tx = x.destination_hash().unwrap();
-					Parent::AlreadySpent { other_tx }
-				}
-				None => match self.stempool.get_blockchain_spent(&output_ref.commit) {
-					Some(x) => {
-						let other_tx = x.destination_hash().unwrap();
-						Parent::AlreadySpent { other_tx }
-					}
-					None => Parent::BlockTransaction,
-				},
-			}
-		})
-	}
-
-	// search_pool_spents is the second half of pool input detection, after the
-	// available_outputs have been checked. This returns either a
-	// Parent::AlreadySpent or None.
-	fn search_pool_spents(&self, output_commitment: &Commitment) -> Option<Parent> {
-		self.pool
-			.get_internal_spent(output_commitment)
-			.map(|x| Parent::AlreadySpent {
-				other_tx: x.destination_hash().unwrap(),
-			})
-	}
-
-	// search_pool_spents is the second half of pool input detection, after the
-	// available_outputs have been checked. This returns either a
-	// Parent::AlreadySpent or None.
-	fn search_stempool_spents(&self, output_commitment: &Commitment) -> Option<Parent> {
-		self.stempool
-			.get_internal_spent(output_commitment)
-			.map(|x| Parent::AlreadySpent {
-				other_tx: x.destination_hash().unwrap(),
-			})
-	}
-
-	/// Get the number of transactions in the stempool
-	pub fn stempool_size(&self) -> usize {
-		self.stempool.num_transactions()
-	}
+	// Aggregate this new tx with all existing txs in the pool.
+	// If we can validate the aggregated tx against the current chain state
+	// then we can safely add the tx to the pool.
+	pub fn add_to_pool(&mut self, tx: &Transaction) -> Result<(), PoolError> {
+		let mut txs = self.transactions
+			.iter()
+			.map(|x| *x.clone())
+			.collect::<Vec<_>>();
+		txs.push(tx.clone());
+		let agg_tx =
+			transaction::aggregate_with_cut_through(txs).map_err(|_| PoolError::GenericPoolError)?;
```

Comment:
Missing ```self.transactions.push(tx.clone())```? Don't want to bug you since it's a WIP ^^but figure I would write it anyway.